src/ConnectAuth.ts

```diff
@@ -0,0 +1,192 @@
+import { AuthError, AuthErrorCode } from './AuthError';
+import { decodeUnverifiedConnectJwt, verifyConnectJwt } from './Jwt';
+import { RequestReader } from './requestReader';
+import { ConnectJwt, ContextQsh, CredentialsLoader, InstallType } from './types';
+
+interface CommonVerifyArgs<E> {
+  baseUrl: string;
+  requestReader: RequestReader;
+  loadCredentials: CredentialsLoader<E>;
+  skipQueryStringHashCheck?: boolean; // Bitbucket does not implement Query String Hash
+}
+
+export type VerifyInstallArgs<E> = CommonVerifyArgs<E>;
+
+export interface VerifyInstallNewResponse {
+  type: InstallType.newInstall;
+  clientKey: string;
+}
+
+export interface VerifyInstallUpdateResponse<E> {
+  type: InstallType.update;
+  clientKey: string;
+  connectJwt: ConnectJwt;
+  storedEntity: E;
+}
+
+export type VerifyInstallResponse<E> = VerifyInstallNewResponse | VerifyInstallUpdateResponse<E>;
+
+export interface VerifyRequestArgs<E> extends CommonVerifyArgs<E> {
+  useContextJwt?: boolean;
+}
+
+export interface VerifyRequestResponse<E> {
+  connectJwt: ConnectJwt;
+  storedEntity: E;
+}
+
+export interface VerifyQueryStringHashArgs {
+  baseUrl: string;
+  connectJwt: ConnectJwt;
+  requestReader: RequestReader;
+  useContextJwt?: boolean;
+  skipQueryStringHashCheck?: boolean; // Bitbucket does not implement Query String Hash
+}
+
+/**
+ * Entry point for Connect request authentication.
+ */
+export class ConnectAuth {
```

Comment:
New class name. Naming is hard but I'm trying to to make it explicit that it deals with Connect Authorization and is not a generic Addon or App helper.

Reply:
Since this class doesn't have any state anymore, may it be dropped at all in favor of regular functions?